### ISSUE TITLE
`#[test]` creates some bugs

### ISSUE THREAD

**Issue by neonaot** [intellij-rust/intellij-rust#8129]
## Environment

* **IntelliJ Rust plugin version:** 0.4.161.4249-213-nightly
* **Rust toolchain version:** 1.58.0-nightly (547a6ffee 2021-10-21) x86_64-apple-darwin
* **IDE name and version:** CLion 2021.3 Beta (CL-213.5744.5)
* **Operating system:** macOS 11.4
* **Macro expansion engine:** new
* **Name resolution engine:** new

## Problem description
 

- [ ] Plugin does not detect E0601:

<img width="587" alt="изображение" src="https://user-images.githubusercontent.com/49211026/142443234-bce3d6f3-5d0d-4c27-926b-4e189e9f444e.png">

- [ ] Detect `A value named _foo_ has already been defined in this module [E0428]`, but code is ok

<img width="662" alt="изображение" src="https://user-images.githubusercontent.com/49211026/142444211-490c2bfb-ea5e-4783-9137-086a8aee048b.png">

## Steps to reproduce
1. 
```rust 
#[test]
fn main(){
}

```

2. 
```rust 
#[test]
fn foo(){
}

fn main() {}
fn foo() {}
```
